client/ios/dropin/Components/EventFormView.swift

```diff
@@ -0,0 +1,237 @@
+//
+//  EventFormView.swift
+//  dropin
+//
+//  Created by Michael Voemel on 15.04.2025.
+//
+
+import SwiftUI
+import MapKit
+
+struct MapLocation: Identifiable {
+    let id: Int
+    let coordinate: CLLocationCoordinate2D
+}
+
+struct DropInEventFormView: View {
+    @Binding var event: DropInEvent
+    @FocusState private var focusedField: Field?
+    @State private var region: MKCoordinateRegion
+    @State private var locationText: String = ""
+    
+    var onSave: () -> Void
+    
+    enum Field: Hashable {
+        case title, description, location
+    }
+    
+    init(event: Binding<DropInEvent>, onSave: @escaping () -> Void) {
+        self._event = event
+        self.onSave = onSave
+        
+        // Defaults to Zurich
+        let coordinate = CLLocationCoordinate2D(
+            latitude: event.wrappedValue.latitude != 0 ? event.wrappedValue.latitude : 47.3769,
+            longitude: event.wrappedValue.longitude != 0 ? event.wrappedValue.longitude : 8.5417
+        )
+        
+        self._region = State(initialValue: MKCoordinateRegion(
+            center: coordinate,
+            span: MKCoordinateSpan(latitudeDelta: 0.05, longitudeDelta: 0.05)
+        ))
+        
+        if event.wrappedValue.latitude != 0 && event.wrappedValue.longitude != 0 {
+            // TODO: implement reverseGeocodeLocation function to get location name from coord
+            self._locationText = State(initialValue: "Location set")
+        } else {
+            self._locationText = State(initialValue: "")
+        }
+    }
+    
+    var body: some View {
+        NavigationStack {
+            Form {
+                Section {
+                    TextField("Title", text: $event.title)
+                        .font(.title)
+                        .focused($focusedField, equals: .title)
```

Comment:
maybe make title bold for nicer UI/UX

```suggestion
                    TextField("Title", text: $event.title)
                        .font(.title)
                        .bold()
                        .focused($focusedField, equals: .title)
```